src/metabase/search/impl.clj

```diff
@@ -553,14 +545,17 @@
                                     (empty?
                                      (remove matching-columns search.config/displayed-columns)))
                            (match-context-thunk))
-         :collection     (merge {:id              collection_id
-                                 :name            collection_name
-                                 :authority_level collection_authority_level
-                                 :type            collection_type}
-                                (when effective_parent
-                                  effective_parent)
-                                (when collection_effective_ancestors
-                                  {:effective_ancestors collection_effective_ancestors}))
+         :collection     (if (and trashed_directly (not= "collection" model))
+                           (select-keys (collection/trash-collection)
+                                        [:id :name :authority_level :type])
+                           (merge {:id              collection_id
+                                   :name            collection_name
+                                   :authority_level collection_authority_level
+                                   :type            collection_type}
+                                  (when effective_parent
+                                    effective_parent)
```

Comment:
this was my own code, but I just realized that we don't need the `when` since if `effective_parent` is nil it'll be ignored by the merge 😀 
```suggestion
                                  ;; for  non-root collections, override :collection with the values for its effective parent
                                  effective_parent
```